gloo/test/send_recv_test.cc

```diff
@@ -235,6 +235,48 @@ TEST_P(SendRecvTest, RecvFromAny) {
   });
 }
 
+TEST_P(SendRecvTest, AbortRecv) {
+  const auto transport = std::get<0>(GetParam());
+  const auto contextSize = std::get<1>(GetParam());
+  constexpr uint64_t slot = 0x1337;
+  spawn(transport, contextSize, [&](std::shared_ptr<Context> context) {
+    // test that an unbounded buffer should be able to abort its own waitRecv.
+    bool recvCompleted;
+    if (context->rank == 0) {
+      int tmp;
+      auto buf = context->createUnboundBuffer(&tmp, sizeof(tmp));
+
+      std::thread waitRecvThread([&]() {
+        std::vector<int> ranks = {1};
+        int srcRank = -1;
+        buf->recv(ranks, slot);
+        recvCompleted = buf->waitRecv(&srcRank);
```

Comment:
Not quite understand the concepts from gloo.

What's the purpose of calling `buf->recv(..)` first?

Reply:
From what I understand, `buf->recv` will call the `recv` function from a particular `pair` (basically an encapsulation of 2 gloo processes that can communicate), which notifies the sender that there is a receive pending, but does not block. Then, `waitRecv` blocks until `Pair::read` indicates that data has been received.

Reply:
I see, receive operation needs a signal first.
And it's similar code to `ProcessGroupGloo::recv` or `ProcessGroupGloo::recvAnysource`.

For normal recv, it's necessary. Otherwise, `RecvWork` will wait for nothing and block forever.

Since you are aborting anyway, I think this line could be removed to make the test subject prominent.